spyder/app/mainwindow.py

```diff
@@ -845,15 +845,6 @@ def pre_visible_setup(self):
         if self.layouts is not None:
             self.layouts.register_custom_layouts()
 
-        # Needed to ensure dockwidgets/panes layout size distribution
-        # when a layout state is already present.
-        # See spyder-ide/spyder#17945
-        if (
-            self.layouts is not None
-            and self.get_conf('window/state', default=None)
-        ):
-            self.layouts.before_mainwindow_visible()
```

Comment:
This bit of code was what actually generated issue #21596 for the layout distribution shown in the OP. That's why I removed it.

However, I checked that issue #17945 was not triggered again by this removal. But please check if that's the case for you too.

Reply:
I was able to reproduce #17945 by launching Spyder from the work on this PR :/ :

![layout_distribution](https://github.com/spyder-ide/spyder/assets/16781833/d0f618af-8241-45f5-be37-5a20e2364213)

|Initial distribution with maximum height given to the IPython Console|Final distribution after closing and launching Spyder multiple times|
|-|-|
|![image](https://github.com/spyder-ide/spyder/assets/16781833/1ca636f5-829f-403c-8001-cd752b625f69)|![image](https://github.com/spyder-ide/spyder/assets/16781833/dde2e871-dcf1-4aae-b406-e24f22783edb)|

Reply:
Ok, it's very weird but I was easily able to reproduce the problem today (I thought I have tested it and it wasn't happening anymore).

However, I amended one of my last commits and now it should be working as expected (both for issue #17945 and #21596). So, please test again.

Reply:
Checked again and seems like indeed the latest changes help with both issues :+1: